src/Clash/Cores/Ethernet/Examples/RxStacks.hs

```diff
@@ -0,0 +1,82 @@
+{-# language FlexibleContexts #-}
+
+{-|
+Module      : Clash.Cores.Ethernet.Examples.RxStacks
+Description : Provides the entire receive stack as a circuit.
+-}
+module Clash.Cores.Ethernet.Examples.RxStacks
+  ( macRxStack
+  , ipRxStack
+  ) where
+
+import Clash.Cores.Crc
+import Clash.Cores.Crc.Catalog
+import Clash.Prelude
+
+import Protocols
+import Protocols.Extra.PacketStream
+import Protocols.Extra.PacketStream.AsyncFIFO ( asyncFifoC )
+import Protocols.Extra.PacketStream.Converters ( upConverterC )
+import Protocols.Extra.PacketStream.Routing ( packetDispatcherC )
+
+import Clash.Cores.Ethernet.IP.IPPacketizers
+import Clash.Cores.Ethernet.IP.IPv4Types
+import Clash.Cores.Ethernet.Mac.EthernetTypes
+import Clash.Cores.Ethernet.Mac.FrameCheckSequence ( fcsValidatorC )
+import Clash.Cores.Ethernet.Mac.MacPacketizers ( macDepacketizerC )
+import Clash.Cores.Ethernet.Mac.Preamble ( preambleStripperC )
+
+-- | Processes received ethernet frames
+macRxStack
+  :: forall
+  (dataWidth :: Nat)
+  (dom :: Domain)
+  (domEth :: Domain).
+  ( HiddenClockResetEnable dom
+  , KnownDomain domEth
+  , HardwareCrc Crc32_ethernet 8 dataWidth
+  )
+  => 1 <= dataWidth
+  => KnownNat dataWidth
+  => Clock domEth
+  -> Reset domEth
+  -> Enable domEth
+  -> Signal dom MacAddress
+  -> Circuit (PacketStream domEth 1 ()) (PacketStream dom dataWidth EthernetHeader)
+macRxStack ethClk ethRst ethEn macAddressS =
+    upConverterC'
+    |> asyncFifoC'
+    |> preambleStripperC
+    |> fcsValidatorC
+    |> macDepacketizerC
+    |> filterMetaS (isForMyMac <$> macAddressS)
+  where
+    upConverterC' = exposeClockResetEnable upConverterC ethClk ethRst ethEn
+    asyncFifoC' = asyncFifoC d4 ethClk ethRst ethEn hasClock hasReset hasEnable
+    isForMyMac myMac (_macDst -> to) = to == myMac || to == broadcastMac
+
+-- | Processes received IP packets
+ipRxStack
+  :: forall
+  (dataWidth :: Nat)
+  (dom :: Domain)
+  (domEth :: Domain).
+  ( HiddenClockResetEnable dom
+  , KnownDomain domEth
+  , HardwareCrc Crc32_ethernet 8 dataWidth
+  )
+  => 1 <= dataWidth
+  => KnownNat dataWidth
```

Comment:
I think this was already in the original RxStack like this, but formatting the constraints half as a tuple looks a bit weird. I'd prefer something like
```haskell
ipRxStack :: forall (dataWidth :: Nat) (dom :: Domain) (domEth :: Domain)
   . HiddenCLockResetEnable dom
  => KnownDomain domEth
  => HardwareCrc Crc32_ethernet 8 dataWidth
  => 1 <= dataWidth
  => KnownNat dataWidth
```
Or format it entirely as a tuple.